### ISSUE TITLE
logan's shadow mission 5 part 1 crash

### ISSUE THREAD

**Issue by Levan7** [hrydgard/ppsspp#10415]
I am using v1.5.4-113-gdc6a4ea75 build on x64 windows 10. At the end of the level when the bridge explodes the ppsspp crashes, funny thing though is that if you try to blow up the concrete blocks yourself and when dx9 is enabled the emulator goes further then open gl and dx 11

![1](https://user-images.githubusercontent.com/5770076/34070210-553865c0-e27b-11e7-86c1-f3267711213a.gif)

Below is the dx9

![2](https://user-images.githubusercontent.com/5770076/34070216-73f2d41e-e27b-11e7-97ab-ce99a8d028cd.gif)

Log files
[logs.zip](https://github.com/hrydgard/ppsspp/files/1564731/logs.zip)

**Comment by LunaMoo** [comment]
The square shape in smoke looks incorrect as well, reminds me of #10165 ~ Resistance Retribution which on a side note is also a very crashy game, maybe those games are using same engine. In which case this could also be related to #4626.

**Comment by Levan7** [comment]
@LunaMoo 
Yes those to games are done by same developer and use the same engine.
I think there are box shaped water droops as well

**Comment by unknownbrackets** [comment]
Did this work in some previous version, such as 1.4.x or 1.2.x?  Or has it always crashed here?

-[Unknown]

**Comment by Levan7** [comment]
@unknownbrackets 
Well it always used to happen

**Comment by benderscruffy** [comment]
does it still crash on mission 5 part 1 anyone have a save

**Comment by murtisoft** [comment]
Savegame at mission 5 part 1 attached;
[UCUS98606SyphonFilterLogansShadow.zip](https://github.com/hrydgard/ppsspp/files/5146147/UCUS98606SyphonFilterLogansShadow.zip)

**Comment by vegitafc** [comment]
Hi murtisoft. The only solution i found was to search for a complete savestate and complete the rest of the missions.
Savestates: https://gamefaqs.gamespot.com/psp/939487-syphon-filter-logans-shadow/saves.

**Comment by murtisoft** [comment]
It doesnt bother me much that I cant progress. I was more interested in helping you guys fix problems.
Thanks anyway.

**Comment by Kayael76** [comment]
So I arrived at that point in game,  v1.10.3 on W10, i7 4790k, GTX 970 and OpenGL backend. It doesn't crash while windowed and certainly does a weird thing, but not always. However it indeed freezes at the end of the cutscene (the second gif in the OP), though it's not gamebreaking since you can skip the cutscene entirely.

![Sans titre 2](https://user-images.githubusercontent.com/48800201/98416172-e4b57f00-207e-11eb-9131-92e87666b1bc.jpg)

If i click "Yes", it keeps playing. I had this message only once.

I recorded a GE dump at that point : 

[recording.zip](https://github.com/hrydgard/ppsspp/files/5503577/recording.zip)

If you look away from the explosion and the shake effect that follows, there's absolutely no problem. It only happens if it's drawn on screen.

**Comment by hrydgard** [comment]
when testing, please use the newest build from https://buildbot.orphis.net/ppsspp/ . There have been improvements to error messages so there's a bigger chance of getting something meaningful from that kind of thing.

But I don't think that rendering is the problem, it's likely some kind of memory corruption, can be timing or just wrong emulation .. these are hard to track down.

**Comment by Panderner** [comment]
Still crashes in v1.10.3-1375-g52c6169a5 for me.
![Screenshot_2020-12-21-19-12-27-184_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/102771350-bc7ca800-43c0-11eb-8a14-283a6d2d3bb1.jpg)

**Comment by SaberKinniku** [comment]
Wanted to report this crash is still happening in version 1.11. I'm using Windows 10

**Comment by Cloud-Hopper** [comment]
> However it indeed freezes at the end of the cutscene (the second gif in the OP), though it's not gamebreaking since you can skip the cutscene entirely.

Unfortunately you can not skip the cutscene on the first playthrough, only in mission mode after you have finished the level already, or in story mode after you have already beaten the story once. So without a gamesave after this level has been beaten it is gamebreaking.

> If you look away from the explosion and the shake effect that follows, there's absolutely no problem. It only happens if it's drawn on screen.

My game still crashed even when I run into the canyon behind and run away. But when I do run away and look away it at least seems to give me the blue screen more consistently, if looking at the explosion or near the tank even if looking away PPSSPP often just freezes completely

I am playing PPSSPP on a Steam Deck using the Decky Loader, PPSSPP v1.14.4. Same issue in both Game mode and desktop mode with PPSSPP in a window. I have tried OpenGL, Vulkan, and Software Rendering. I have tried with both the default auto settings and and a variety of settings to mimic the original PSP. 

I can sometimes get to the part where the tank drops with many graphical glitches, but it usually just crashes as soon as the bridge starts to blow and the tank starts falling before the camera changes to the tank dropping. But beyond that always either gives a blue screen or just freezes PPSSPP completely. Here are a couple of blue screen screenshots from both OpenGL and Vulkan respectively, please let me know if I can provide anything else.

The only proper workaround I could really think of is maybe a modded gamesave where the game has been beaten but all stats reset. Then you could at least skip the cutscene in a Story mode playthrough with fresh stats for unlockables.
![Screenshot_20230119_200707](https://user-images.githubusercontent.com/123131109/213601413-caaff47e-271d-4f56-8ff2-b213824680c4.png)
![Screenshot_20230119_201013](https://user-images.githubusercontent.com/123131109/213601436-fe071c1d-9352-4d6d-ae1b-e9236eef608b.png)

**Comment by LeyoSFO** [comment]
Reporting @hrydgard , this crash still happens in Logan's Shadow (USA & EUR) on PPSSPP 1.15.4

**Comment by stg43x** [comment]
still not fixed???

**Comment by mejddorbez** [comment]
Still happened in the last version

**Comment by sebastian1996** [comment]
I think know what is the issue. You need go to ppsspp.ini file ( memstick\PSP\SYSTEM ) and change the DisplayRefreshRate from 60 to 45. replay the mission and it would pass the cutscene, at least this work for me.

**Comment by Panderner** [comment]
A High FPS can crash Syphon Filter: Logan's Shadow after ending cutscene of mission 5 part 1, 30 FPS cap cheat needed to pass that mission.

30 FPS:

https://github.com/user-attachments/assets/cb20e8cb-a2a8-48d4-b391-a5ef1515cf26

60 FPS:

https://github.com/user-attachments/assets/521d0165-7c09-4961-bab6-7c3cef395f9a

**Comment by hrydgard** [comment]
@Panderner That's good info, thanks! Another case of us running too fast breaking a game :/ PSP performance is hard to model accurately...

Patching this in any reasonable way will be tricky, I think..

**Comment by hubofficiel** [comment]
> Резервная копия миссии 5, часть 1, прилагается; [UCUS98606SyphonFilterLogansShadow.zip](https://github.com/hrydgard/ppsspp/files/5146147/UCUS98606SyphonFilterLogansShadow.zip)

  How do I save it?

**Comment by hrydgard** [comment]
Just adding a video and another savegame I got in a report.

Still we need to find a proper solution here...

[UCUS986060000.zip](https://github.com/user-attachments/files/25163178/UCUS986060000.zip)

https://github.com/user-attachments/assets/0362a7d9-5e1d-467a-bd41-b4fe63a69cfb
